src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_ForEachStatement.cs

```diff
@@ -963,5 +1203,29 @@ private void InstrumentForEachStatement(BoundForEachStatement original, ref Boun
                 result = _instrumenter.InstrumentForEachStatement(original, result);
             }
         }
+
+        /// <summary>
+        /// Produce a while(true) loop
+        ///
+        /// <![CDATA[
+        /// still-true:
+        /// /* body */
+        /// goto still-true;
+        /// ]]> 
+        /// </summary>
+        private static BoundStatement MakeWhileTrueLoop(BoundForEachStatement loop, BoundBlock body)
+        {
+            Debug.Assert(loop.EnumeratorInfoOpt.IsAsync);
+            SyntaxNode syntax = loop.Syntax;
+            GeneratedLabelSymbol startLabel = new GeneratedLabelSymbol("still-true");
+
+            // still-true:
+            // /* body */
+            // goto still-true;
+            return BoundStatementList.Synthesized(syntax, hasErrors: false,
+                 new BoundLabelStatement(syntax, startLabel),
```

Comment:
I think the label needs to be wrapped in a sequence point. Otherwise debugger will highlight the statement before when hitting the label. Look at `RewriteForStatementWithoutInnerLocals` for example.